Userland/Libraries/LibDSP/Track.cpp

```diff
@@ -92,6 +99,25 @@ void Track::current_signal(FixedArray<Sample>& output_signal)
     VERIFY(output_signal.size() == source_signal->get<FixedArray<Sample>>().size());
     // The last processor is the fixed mastering processor. This can write directly to the output data. We also just trust this processor that it does the right thing :^)
     m_track_mastering->process_to_fixed_array(*source_signal, output_signal);
+
+    bool false_variable = false;
+    if (m_sample_lock.compare_exchange_strong(false_variable, true)) {
+        AK::TypedTransfer<Sample>::copy(m_cached_sample_buffer.data(), output_signal.data(), m_cached_sample_buffer.size());
+    }
+    m_sample_lock.store(false);
```

Comment:
The unlock must only happen within the if, since otherwise you’re taking someone else’s lock.